packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -316,10 +316,17 @@ export class ProfilesCache {
     }
 
     // This will retrieve the base profile from imperative
-    public async fetchBaseProfile(): Promise<imperative.IProfileLoaded | undefined> {
+    public async fetchBaseProfile(profileName?: string): Promise<imperative.IProfileLoaded | undefined> {
         const mProfileInfo = await this.getProfileInfo();
         const baseProfileAttrs = mProfileInfo.getDefaultProfile("base");
-        if (baseProfileAttrs == null) {
+        const configApi = mProfileInfo.getTeamConfig().api;
+        if (
+            profileName?.includes(".") &&
+            (baseProfileAttrs == null || !configApi.secure.securePropsForProfile(baseProfileAttrs.profName).includes("tokenValue"))
```

Comment:
It might just be me, but it took me a bit longer to understand what was happening here.

Would it be worth adding one or two lines describing this? 
Or maybe just a link to the diagram (in https://github.com/zowe/zowe-explorer-vscode/issues/2264#issuecomment-2236914511 - and highlight the path Yes-No-Yes 😋 )

Reply:
Added comment here